internal/kafka/providers.go

```diff
@@ -22,6 +22,20 @@ import (
 	"github.com/goava/di"
 )
 
+func DataPlaneConfigProviders() di.Option {
+	// do not inject gcp config when running integration test to avoid accidental cluster creation
+	if environments2.GetEnvironmentStrFromEnv() == environments2.IntegrationEnv {
```

Comment:
This is not correct.
The issue does not only happen when running integration environment but also when running integration tests with the development environment set. This is common when developing locally and wanting to run the integration tests against real OCM, and also on our nightly tests.

I also think that the enablement of this should probably be done at the test suite level and not in the DI framework side, as in some tests we might want to have the GCP configuration provider set

Reply:
Makes sense. I'm thinking to remove the gcp provider in `NewKafkaHelperWithHooks` which is used in the integration tests.
The Injection test is a special case though and I'll have to find a different fix for that.